### PR TITLE
Move tutorial & configurations docs to an mdbook

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
What this PR does:

- Moves the GIR tutorial that used to be part of the website to here so it can be kept up to date whenever changes happens to gir itself
- Move most of the README to proper sections in the book, hopefully in the future we can split into multiple further sections to simplify things for the end-user

The PR doesn't do any "huge" fixes other than some markup/typo fixes and moving things around. An update to the tutorial has to be done later by someone else

Here's the table of content as is right now
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,702 +1,7 @@
...
-
-```toml
-[crate_name_overrides]
-"web_kit2_web_extension" = "webkit2_webextension"
-```
-
-## GIR format reference
-
-It can always be useful to look at the [reference](https://gi.readthedocs.io/en/latest/annotations/giannotations.html) or [schema](https://gitlab.gnome.org/GNOME/gobject-introspection/blob/master/docs/gir-1.2.rnc), right?
+A work in progress book to help with learning how to use `gir` along with a tutorial are available at <https://gtk-rs.org/gir/book>
```
Comment (GuillaumeGomez) [inline README.md:7]: Please add a dot at the end of the sentence.

File: book/book.toml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[book]
+authors = ["Bilal Elmoussaoui"]
```
Comment (GuillaumeGomez) [inline book/book.toml:2]: Please use the same that we use everywhere ("gtk-rs developers").

File: book/src/config_name_override.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
```
Comment (GuillaumeGomez) [inline book/src/config_name_override.md:1]: Please remove this empty line.

**Review by GuillaumeGomez — COMMENTED** [review]
A few nits but otherwise looks good to me. Please fix them and then we can merge. Thanks a lot for working on this!

File: book/book.toml (hunk 2 above)
Comment (bilelmoussaoui) [reply book/book.toml:2]: Ah sorry, completely missed this one

**Comment by bilelmoussaoui** [comment]
@GuillaumeGomez should be ready now

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

**Comment by sdroege** [comment]
This is great, thanks :)

**Pull request merged**